pkg/orchestrator/job-manager.go

```diff
@@ -88,20 +92,48 @@ func (o *Orchestrator) waitForResult(data *scanData) {
 		log.Infof("Image scanned result has arrived. image=%v", data.imageName)
 	case <-ticker.C:
 		log.Warnf("job was timeout. image=%v", data.imageName)
+		o.Lock()
+		data.success = false
+		data.timeout = true
+		data.completed = true
+		o.Unlock()
 	}
 }
 
-func (o *Orchestrator) runJob(data *scanData) error {
+func (o *Orchestrator) runJob(data *scanData) (*batchv1.Job, error) {
 	job := o.createJob(data)
 	log.Debugf("Created job=%+v", job)
 
 	log.Infof("Running job %s/%s to scan image %s", job.GetNamespace(), job.GetName(), data.imageName)
 	_, err := o.clientset.BatchV1().Jobs(job.GetNamespace()).Create(job)
 	if err != nil {
-		return fmt.Errorf("failed to create job: %v/%v. %v", job.GetName(), job.GetNamespace(), err)
+		return nil, fmt.Errorf("failed to create job: %v/%v. %v", job.GetName(), job.GetNamespace(), err)
 	}
 
-	return nil
+	return job, nil
+}
+
+func (o *Orchestrator) deleteJobIfNeeded(job *batchv1.Job, isSuccessfulJob bool) {
+	switch o.scanConfig.DeleteJobPolicy {
+	case config.DeleteJobPolicyAll:
+		o.deleteJob(job)
+
+	case config.DeleteJobPolicySuccessful:
+		if isSuccessfulJob {
+			o.deleteJob(job)
+		}
+	}
+}
+
+func (o *Orchestrator) deleteJob(job *batchv1.Job) {
+	dpb := metav1.DeletePropagationBackground
+	deleteOptions := &metav1.DeleteOptions{PropagationPolicy: &dpb}
+
+	log.Infof("Deleting job %s/%s", job.GetNamespace(), job.GetName())
+	err := o.clientset.BatchV1().Jobs(job.GetNamespace()).Delete(job.GetName(), deleteOptions)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Errorf("failed to delete job: %s/%s. %v", job.GetName(), job.GetNamespace(), err)
```

Comment:
It's a common way to print (saw it on Istio I think), and this is not the first place we are printing like this here. I think it will be clear to distinguish what is namespace and what is name.